### PR TITLE
Manage categories with FireStore (loading data)

### PR DESCRIPTION
**Pull request by ramshty**
### Summary
This PR introduces and integrates the categories into the backend, enabling fetching and loading them in the application. The integration includes:

- **Function Creation in Search ViewModel**: Added a new function to the `SearchViewModel` to manage the retrieval and handling of categories.
- **Repository Update**: Extended the repository to include direct support for fetching categories from the backend.

### Details
- A new backend endpoint was integrated to provide a list of categories.
- The `SearchViewModel` now includes a function that fetches and prepares categories for UI consumption.
- Updated repository to handle API calls related to category data, ensuring efficient loading and access within the application.

### Testing
- Verified category fetching and loading functionality.
- Confirmed that categories load correctly in the UI and respond accurately to user actions.

### Notes
This integration lays the groundwork for enhanced search and filtering capabilities by providing category data directly in the `SearchViewModel` and repository.

### CODE REVIEW

**Review by Autek — COMMENTED** [review]
This is a great addition! Storing worker categories in the database is a smart choice as it allows for dynamic management and retrieval of categories without requiring app updates. This setup enables more flexible scaling if new categories or subcategories are needed, and it improves consistency across the app since category data is centralized. It also streamlines any future localization efforts and helps ensure that users always see the latest information. Well done!

File: dummy_credentials.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (JitterKiller) [inline dummy_credentials.json:1]: I wonder what this json is for, maybe an oversight?

File: app/src/main/java/com/arygm/quickfix/model/search/SearchViewModel.kt
Comment (JitterKiller) [inline app/src/main/java/com/arygm/quickfix/model/search/SearchViewModel.kt:1]: Great Work with the ViewModel!

**Review by JitterKiller — APPROVED** [review]
Looks great! The integration of categories into the backend is well-implemented, and the updates to the SearchViewModel and repository make it easy to fetch and handle category data effectively. Approved! 👍

**Pull request merged**